flyteadmin/pkg/common/executions.go

```diff
@@ -3,17 +3,24 @@ package common
 import (
 	"fmt"
 
+	"github.com/wolfeidau/humanhash"
 	"k8s.io/apimachinery/pkg/util/rand"
 
 	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/core"
 )
 
 const ExecutionIDLength = 20
+const ExecutionIDLengthLimit = 63
 const ExecutionStringFormat = "a%s"
 
 /* #nosec */
-func GetExecutionName(seed int64) string {
+func GetExecutionName(seed int64, enableHumanHash bool) string {
 	rand.Seed(seed)
+	if enableHumanHash {
+		hashKey := []byte(rand.String(ExecutionIDLength))
+		result, _ := humanhash.Humanize(hashKey, 3)
```

Comment:
should we handle the error here?
```suggestion
		result, err := humanhash.Humanize(hashKey, 3)
```